cmd/tuf/app/init.go

```diff
@@ -321,7 +324,8 @@ func ClearEmptySignatures(store tuf.LocalStore, role string) error {
 }
 
 func jsonMarshal(v interface{}) ([]byte, error) {
-	b, err := cjson.Marshal(v)
+	// We don't need to canonically encode the payload in the store.
+	b, err := json.Marshal(v)
```

Comment:
When changing from cjson to `encoding/json` for Marshalling, there is no risk that the checksum/signature for the metadata file will differ from what's written in the snapshot file? Or is this only verified prior to writing to the metadata cache, my memory is failing me here :)

Reply:
Good question! 
On get: we verify the checksum/length based on the retrieved metadata from the store. (no canonicalization)
On write: we write the snapshot file metadata (checksum/length) based on the retrieved metadata from the store (no canonicalization)

Admittedly, I feel like this is generally a little fragile. It requires that the implementations for `GetMeta` on the repo and client's store implementation and the repository writers `SetMeta` implementation return equivalent bytes.

It only works out this way because they both do not canonicalize. So snapshot checksums and client's verification of the checksum will still be correct, since they checksum exactly what we write to the metadata.

Does that make sense? I guess the tldr is no: snapshot checksums are generated based on what we write to the store, regardless of canonicalization or not.

Reply:
> Does that make sense? I guess the tldr is no: snapshot checksums are generated based on what we write to the store, regardless of canonicalization or not.

I think it does! But, this feels like it may break for sigstore tuf client v2, where multiple languages can share cache, as JSON serialization may differ? Although out of scope for this PR, we should keep track of that, right?

Reply:
> But, this feels like it may break for sigstore tuf client v2, where multiple languages can share cache, as JSON serialization may differ?

I *think* things will work by accident, as long as clients `SetMeta` with just the raw bytes, and don't store in a different representation. 

>  we should keep track of that, right?

I think so. We should be able to catch issues with this with the presubmits, but I would like to ask on theupdateframework/specification for x-lang guidance here. Creating an issue